src/Compilers/CSharp/Portable/Binder/ImportChain.cs

```diff
@@ -113,11 +113,11 @@ private ImmutableArray<Cci.UsedNamespaceOrType> TranslateImports(Emit.PEModuleBu
                         var assemblyRef = TryGetAssemblyScope(ns, moduleBuilder, diagnostics);
                         usedNamespaces.Add(Cci.UsedNamespaceOrType.CreateNamespace(ns.GetCciAdapter(), assemblyRef, alias));
                     }
-                    else if (!target.ContainingAssembly.IsLinked)
+                    else if (target is NamedTypeSymbol { ContainingAssembly.IsLinked: false } namedType)
```

Comment:
>NamedTypeSymbol [](http://example.com/codeflow?start=39&length=15)

This is a change from the previous code that expected `TypeSymbol` only. Consider using `is TypeSymbol { ContainingAssembly.IsLinked: false } type` instead.

Reply:
i don't think this will work.  things like Pointers/Arrays don't have containing assemblies.  The old code worked because it would only ever namespace or named types heres.  and this branch was only for the named-type case.  So i'm preserving that same concept here (which again is the only way to have a linked containing assembly afacit).